src/shared/guards/health-check.guard.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { CanActivate } from '@angular/router';
+import { Observable, of } from 'rxjs';
+import { catchError, switchMap } from 'rxjs/operators';
+import { HealthCheckService } from '../services/health-check.service';
+import * as fromRoot from '../../app/store';
+import { Store } from '@ngrx/store';
+
+
+@Injectable()
+export class HealthCheckGuard implements CanActivate {
+    constructor(
+        private healthCheck: HealthCheckService,
+        private store: Store<fromRoot.State>,
+    ) {
+    }
+
+    canActivate() {
+        return this.checkStore().pipe(
```

Comment:
you re not checking the store so maybe you all this differently

Reply:
Changed function name